### PR TITLE
Use PRDraft emoji when the pr is a draft pr for the .issue|.pr command

### LINKED ISSUE #661: Use a grey :PRDraft: emoji for the .pr command if the PR is a draft
## Description

Currently, the `.issue` | `.pr` command uses a green open pull request emoji when showing open pull requests. Although, when pull requests are drafts in Github, then the colour of the sign will be grey. For example, #623 is a draft PR, and doing `.pr 623` currently looks like this:
green-pr-emoji

However, it really should look like this: 
grey-pr-emoji

## Reasoning

I think it would make more sense when it is a draft pr that it should be grey instead of green.

## Proposed Implementation

After having a :PRDraft: emoji, check if `json_data["draft"]` is `True`, and if it is, use the :PRDraft: emoji.

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by ToxicKidz**
## Relevant Issues

Closes #661

## Description

In this PR, I changed the url for getting the pull info to `https://api.github.com/repos/{user}/{repository}/pulls/{number}`, and check if `pull_data["draft"]` is true, and if so, use the :PRDraft: emoji.

## Reasoning

It makes more sense when the PR is a draft that the bot will show the draft PR icon/

## Screenshots

draft-pr-emoji

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] If dependencies have been added or updated, run `pipenv lock`?
- [x] **Lint your code** (`pipenv run lint`)?
- [x] Set the PR to **allow edits from contributors**?

### CODE REVIEW

**Review by Kronifer — APPROVED** [review]
Looks good to me! :shipit:

**Comment by Akarys42** [comment]
Thabsk for the PR! I'll retarget this PR to #642 to generate conflicts only once :P sorry, you may have to solve some

**Review by ChrisLovering — APPROVED** [review]
Looks good to me, tested locally too.

I also tested the `WonderTwins` and `ConvoStarters` cogs since those are the only ones that use PyYAML (Since this PR upgrades the version to latest)

File: bot/exts/evergreen/issues.py [hunk 1]
```diff
@@ -151,12 +151,15 @@ async def fetch_issues(
...
-            async with self.bot.http_session.get(merge_url) as m:
-                if json_data.get("state") == "open":
+            log.trace(f"PR provided, querying GH pulls API for additional information: {pulls_url}")
+            async with self.bot.http_session.get(pulls_url) as p:
+                pull_data = await p.json()
+                if pull_data["draft"]:
+                    emoji = Emojis.pull_request_draft
+                elif pull_data["state"] == "open":
                     emoji = Emojis.pull_request
                 # When the status is 204 this means that the state of the PR is merged
```
Comment (Akarys42) [inline bot/exts/evergreen/issues.py:161]: This comment needs to be updated

**Review by Akarys42 — CHANGES_REQUESTED** [review]
Just an outdated comment

**Pull request merged**